shouldupdate.js

```diff
@@ -172,18 +138,30 @@ function factory (methods) {
    */
   function isEqualProps (value, other) {
     return isEqual(value, other, function (current, next) {
-      if (_isCursor(current) && _isCursor(next)) {
+      if (current == next) {
+        return true;
+      }
+
+      var isCurrentCursor = _isCursor(current);
+      var isNextCursor = _isCursor(next);
+
+      if (isCurrentCursor && isNextCursor) {
         return _isEqualCursor(current, next);
       }
-      if (_isCursor(current) || _isCursor(next)) {
+      if (isCurrentCursor || isNextCursor) {
         return false;
       }
-      if (_isImmutable(current) && _isImmutable(next)) {
-        return current === next;
+
+      var isCurrentImmutable = _isImmutable(current);
+      var isNextImmutable = _isImmutable(next);
+
+      if (isCurrentImmutable && isNextImmutable) {
+        return current.equals(next);
```

Comment:
The same goes here. We need some hard numbers to know if this is the way to go or not.